dali/operators/audio/nonsilence_op_test.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <utility>
+#include <random>
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+
+namespace dali {
+namespace testing {
+
+class NonsilenceOpTest : public ::testing::Test {
+ protected:
+  void SetUp() final {
+    impl_.nsamples_ = 1;
+  }
+
+
+  std::vector<float> input_{0, 0, 0, 0, 1000, -1000, 1000, 0, 0, 0};
+  int window_size_ = 3;
+  std::vector<float> mms_ref_{0, 0, 333333.344, 666666.688, 1000000, 666666.688, 333333.344, 0};
+  std::pair<int, int> nonsilence_region_{2, 5};
+  int buffer_length_ = 10;
+  TensorShape<1> shape_ = {buffer_length_};
+  NonsilenceOperatorCpu::Impl impl_;
```

Comment:
I think you should instantiate the actual operator.

Reply:
This is a test of operator's implementation. Actual operator (with its boiler plate and instantiation) is tested in python tests.
I purposely split this into implementation class. This makes the code testable == better.